iOS/Collection View Management /CollectionViewDataSource.swift

```diff
@@ -93,13 +100,226 @@ extension CollectionViewDataSource {
     }
 }
 
+// MARK: - Snapshot Convenience Functions
+
+extension CollectionViewDataSource {
+
+    // Synchronous Functions
```

Comment:
Synchronous versions can be "fire-and-forget". Note that none of these have completion handlers. Any completion handlers should be handled using async functions below.